models/classes/search/ResultSetResponseNormalizer.php

```diff
@@ -70,18 +77,43 @@ public function normalize(SearchQuery $searchQuery, ResultSet $resultSet, string
                     $content['label'] = __('Access Denied');
                 }
 
+                $resource = $this->getResource($content['id'] ?? '');
+
+                $readonly = false;
+
+                $topLevelClass = $this->getClass(TaoOntology::CLASS_URI_OBJECT);
+
+                foreach ($resource->getTypes() as $type) {
+                    $accessibleResources = $permissionHelper
+                    ->filterByPermission(
+                        [$type->getUri()],
+                        PermissionInterface::RIGHT_READ
+                    );
+
+                    if (empty($accessibleResources) || !$isAccessible) {
+                        $readonly = true;
+                        break;
+                    }
+
+                    $class = $this->getClass($type->getUri());
+                    $readonly = $this->checkParentClassPermission($class, $permissionHelper, $topLevelClass);
+
+                    if ($readonly === true) {
+                        break;
```

Comment:
Would it be possible to consolidate https://github.com/oat-sa/tao-core/pull/3157/files#diff-7238f77e35d72a5abbb77e5ab3057370654a655a850c760d84bca645a59d0b4eR106 wihing existing code? 
```suggestion
                    $content['id'] = '';
                                                break;
```